x-pack/plugins/ml/server/models/data_recognizer/modules/apm_transaction/manifest.json

```diff
@@ -1,29 +1,29 @@
 {
   "id": "apm_transaction",
   "title": "APM",
-  "description": "Detect anomalies in transactions from your APM services.",
+  "description": "Detect anomalies in transactions from your APM services for metric data.",
   "type": "Transaction data",
   "logoFile": "logo.json",
-  "defaultIndexPattern": "apm-*-transaction",
+  "defaultIndexPattern": "apm-*-metric-*",
   "query": {
     "bool": {
       "filter": [
-        { "term": { "processor.event": "transaction" } },
-        { "exists": { "field": "transaction.duration" } }
+        { "term": { "processor.event": "metric" } },
+        { "exists": { "field": "transaction.duration.histogram" } }
```

Comment:
You can also use a `term` query here: `metricset.name:"transaction"`. That is slightly more intentful than `exists`. We use `exists` internally for backwards compatibility purposes, but for the ML job, I'm inclined to say it's not necessary, as it should operate only on (fairly) recent data.

Reply:
Does the datafeed not filter at all?

Reply:
No, there is no query in the datafeed config.

Reply:
Should there be one? The metrics index includes a lot of other data/metricsets as well.

Reply:
@dgieselaar the datafeed is gathering statistics on: transaction, error rate, and transactions per minute.

I am not sure that any filtering is required as we are aggregating the data by @timestamp, transaction.type, and service.name and running on the aggregated results.

Reply:
e.g. event.outcome, transaction.type, service.name and @timestamp are all available on span destination metrics. So unless I'm mistaken we need to filter here?

Reply:
Agree, a filter here seems a good idea, both wrt performance and correctness of the produced results.

Reply:
Just to close the loop on this one - the datafeed now contains the query:

```
  "query": {
    "bool": {
      "filter": [
        { "term": { "processor.event": "metric" } },
        { "term": { "metricset.name": "transaction" } }
      ]
    }
  },
```

And when the job is created from the APM UI, a filter for the service environment is added, as done by @dgieselaar in https://github.com/elastic/kibana/pull/111957/commits/1dc720eb1797e764c38691644cd9161f34681438.